### ISSUE TITLE
Can not obtain a signed url for a file with a space in its name

### ISSUE THREAD

**Issue by prapicault** [googleapis/google-cloud-ruby#3064]
#### Environment details
  - OS: MacOS
  - Ruby version: 2.4.4
  - Gem name and version: google-cloud-storage 1.17.0

#### Steps to reproduce
1 - Create a bucket `my-todo-app`
2 - Create a folder called `foo`
3 - Add a file called `file with space.txt`
4 - Generate a signed_url following the code at http://googleapis.github.io/google-cloud-ruby/docs/google-cloud-storage/latest/#Using_Signed_URLs 
5 - Grab the URL generated in the previous step and open it
6 - Notice you get an error saying that the `specified key does not exist`. Also notice that the detailed error message says "No such object: my-todo-app/foo/file+with+space.txt"

#### Code example

```ruby
require "google/cloud/storage"

storage = Google::Cloud::Storage.new

bucket = storage.bucket "my-todo-app"
file = bucket.file "foo/file with space.txt"
shared_url = file.signed_url method: "GET",
                             expires: 300 # 5 minutes from now
```

Thanks!

**Comment by nagarjunac** [comment]
I am facing the same issue.  I have found out the cause for it. 
https://github.com/googleapis/google-cloud-ruby/blob/master/google-cloud-storage/lib/google/cloud/storage/file/signer.rb
On line no 46 => "CGI.escape node" is substituting 'spaces' in filename with '+' instead of unicode char '%20'.
By overriding that line with  "Addressable::URI.escape node" is working for me.

**Comment by prapicault** [comment]
Thanks @nagarjunac. I had noticed that line of code as well and I tried URI.escape without success. Glad you found the `Addressable::URI`
I'll look into providing a patch for this.

**Comment by blowmage** [comment]
How are you pulling the signed url? Is your client replacing the `+` in the path with `%20`?

**Comment by prapicault** [comment]
> How are you pulling the signed url? Is your client replacing the `+` in the path with `%20`?

I'm copying and pasting into chrome.

**Comment by blowmage** [comment]
Is it consistently erring in every client? Does it work for you in Firefox or curl or wget?

**Comment by nagarjunac** [comment]
Yes it is consistently erring in every client. It is not working in Firefox and curl as well.

**Comment by prapicault** [comment]
While researching this, I've also played with the bucket webclient, and the API https://cloud.google.com/storage/docs/json_api/v1/buckets/get and you can notice that the whitespaces are encoded with `%20`, not with `+`. HTH.

**Comment by blowmage** [comment]
This looks like a [regression](https://github.com/googleapis/google-cloud-ruby/pull/1923/files#diff-030a250aa9bc3edca850f0e6158da318L44) introduced in #1923.

Unfortunately we don't have any acceptance tests using spaces to catch this. Here is an acceptance test that fails:

```ruby
  it "file with spaces should create a signed read url" do
    local_file = File.new files[:logo][:path]
    file = bucket.create_file local_file, "some/path with spaces/Cloud Logo Signed Url with spaces.png"

    five_min_from_now = 5 * 60
    url = file.signed_url method: "GET",
                          expires: five_min_from_now

    uri = URI url
    http = Net::HTTP.new uri.host, uri.port
    http.use_ssl = true
    http.ca_file ||= ENV["SSL_CERT_FILE"] if ENV["SSL_CERT_FILE"]

    resp = http.get uri.request_uri
    resp.code.must_equal "200"

    Tempfile.open ["google-cloud", ".png"] do |tmpfile|
      tmpfile.binmode
      tmpfile.write resp.body
      tmpfile.size.must_equal local_file.size

      File.read(local_file.path, mode: "rb").must_equal File.read(tmpfile.path, mode: "rb")
    end
  end
```

**Comment by blowmage** [comment]
This bug affects 1.17.0 and 1.16.0.

**Comment by blowmage** [comment]
There is a fix contributed to fix this in #3081, but we are waiting on a signed CLA to accept.
